libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2093,6 +2093,54 @@ void GCS_MAVLINK::send_hwstatus()
         0);
 }
 
+bool GCS_MAVLINK::try_send_gps_message(const enum ap_message id)
+{
+    AP_GPS *gps = get_gps();
+    if (gps == nullptr) {
+        return true;
+    }
+
+    bool ret = true;
+    switch(id) {
+    case MSG_SYSTEM_TIME:
+        CHECK_PAYLOAD_SIZE(SYSTEM_TIME);
+        send_system_time(*gps);
+        ret = true;
+        break;
+    case MSG_GPS_RAW:
+        CHECK_PAYLOAD_SIZE(GPS_RAW_INT);
+        send_gps_raw(*gps);
```

Comment:
Agreed on the boolean.

I think I'll remove the size-check from here rather than in the function; leaving it here is misleading as in every other case if that check passes there's room for everything in the following called function.  Minor thing...